src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -209,24 +210,25 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 		/* extract index key information from the index's pg_index info */
 		indexInfo = BuildIndexInfo(indexDesc);
 
-		/* only check unique indexes */
-		if (indexInfo->ii_Unique == false)
+		/* only check unique indexes and exclusion constraints. */
+		if (indexInfo->ii_Unique == false && indexInfo->ii_ExclusionOps == NULL)
 		{
 			index_close(indexDesc, NoLock);
 			continue;
 		}
 
 		/*
-		 * Citus cannot enforce uniqueness constraints with overlapping shards. Thus,
-		 * emit a warning for unique indexes on append partitioned tables.
+		 * Citus cannot enforce uniqueness/exclusion constraints with overlapping shards. Thus,
+		 * emit a warning for unique indexes and exclusion constraints on append partitioned tables.
 		 */
 		if (distributionMethod == DISTRIBUTE_BY_APPEND)
 		{
 			ereport(WARNING, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-							  errmsg("table \"%s\" has a unique constraint",
+							  errmsg("table \"%s\" has a unique or EXCLUDE constraint",
 									 distributedRelationName),
-							  errdetail("Unique constraints and primary keys on "
-										"append-partitioned tables cannot be enforced."),
+							  errdetail(
```

Comment:
We have 90 characters hard limit. Error detail seems to pass the limit. Also, we do not start the detail with an empty line. Could you please check other error message formatting in the same file?